JavaFXUserInterface/src/main/java/controllers/AssetsController.java

```diff
@@ -171,9 +177,115 @@ public void attachEvents() {
         addAssetBtn.setOnMouseClicked(mouseEvent -> uiUtilities.changeScene(TextConstants.ADD_ASSETS_SCENE, addAssetBtn.getScene()));
 
         sortingSetUp();
+        filterSetUp();
         searchAssets();
     }
 
+    /**
+     * setup the transitions and the content for the filter options
+     *
+     * @author Paul
+     */
+    private void filterSetUp() {
+        TranslateTransition openNav = new TranslateTransition(new Duration(350), navList);
+        TranslateTransition closeNav = new TranslateTransition(new Duration(350), navList);
+        FadeTransition fadeTransition = new FadeTransition(new Duration(350), navList);
+        navList.setVisible(false);
+        filterAsset.setOnAction((ActionEvent evt) -> {
+            navList.setVisible(true);
+            if (navList.getTranslateX() != 0) {
+                fadeTransition.setToValue(1.0);
+                fadeTransition.setFromValue(0.0);
+                fadeTransition.play();
+                openNav.setToX(0);
+                openNav.play();
+            } else {
+                fadeTransition.setToValue(0.0);
+                fadeTransition.setFromValue(1.0);
+                fadeTransition.play();
+                closeNav.setToX((navList.getWidth()));
+                closeNav.play();
+            }
+        });
+
+        int i = 0;
+        CheckBox option;
+        for (AssetType assetType : assetTypeDAO.getAssetTypeList()) {
+            assetTypeFilterCondition.put(assetType.getName(), false);
+            option = new CheckBox(assetType.getName());
+            option.selectedProperty().addListener((observable, oldValue, newValue) -> {
+                assetTypeFilterCondition.replace(assetType.getName(), newValue);
+                generateContent();
+            });
+            navList.add(option, 0, i);
+            i++;
+        }
+        i = 0;
+        for (ThresholdEnum thresholdEnum : ThresholdEnum.values()) {
+            thresholdFilterCondition.put(thresholdEnum.getValue(), false);
+            option = new CheckBox(thresholdEnum.getValue());
+            option.selectedProperty().addListener((observable, oldValue, newValue) -> {
+                thresholdFilterCondition.replace(thresholdEnum.getValue(), newValue);
+                generateContent();
+            });
+            navList.add(option, 1, i);
```

Comment:
The use of int i could be replaced by automatic call to the index in the enum values, such as: 
ThresholdEnum.valueOf(thresholdEnum.getValue()).ordinal()